### PR TITLE
Drop support for DatabaseOrdinary in MaterializedMySQL

### PR DESCRIPTION
**Pull request by stigsb**
The default database engine has been "Atomic" since ClickHouse 20.10. To improve maintainability of the experimental MaterializedMySQL feature, this PR drops support for the "Ordinary" engine for backing MaterializedMySQL.

Also get rid of thread name logic for StorageMaterializeMySQL wrapping, use setInternalQuery instead (similar to MaterializedPostgreSQL).

Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Drop support for using Ordinary databases with MaterializedMySQL

Detailed description / Documentation draft:

> By adding documentation, you'll allow users to try your new feature immediately, not when someone else will have time to document it later. Documentation is necessary for all features that affect user experience in any way. You can add brief documentation draft above, or add documentation right into your patch as Markdown files in [docs](https://github.com/ClickHouse/ClickHouse/tree/master/docs) folder.

> If you are doing this for the first time, it's recommended to read the lightweight [Contributing to ClickHouse Documentation](https://github.com/ClickHouse/ClickHouse/tree/master/docs/README.md) guide first.

> Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Databases/DatabaseFactory.cpp [hunk 1]
```diff
@@ -211,14 +211,9 @@ DatabasePtr DatabaseFactory::getImpl(const ASTCreateQuery & create, const String
             if (engine_define->settings)
                 materialize_mode_settings->loadFromQuery(*engine_define);
 
-            if (create.uuid == UUIDHelpers::Nil)
```
Comment (tavplubix) [inline src/Databases/DatabaseFactory.cpp:214]: If `create.uuid` is Nil, we should throw exception which explains that it's not supported anymore and how to fix it and make server start (remove metadata file and database dirs)

File: src/Databases/IDatabaseReplicating.h [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace DB
+{
+
+class IDatabaseReplicating
+{
+public:
+    virtual void stopReplication() = 0;
+    virtual ~IDatabaseReplicating() = default;
+};
```
Comment (tavplubix) [inline src/Databases/IDatabaseReplicating.h:11]: I'm not sure that we need separate class for this, maybe just add methods like `hasReplicatingThread` (that returns `false` by default) and `stopReplication()` (that throws `NOT_IMPLEMENTED`/`LOGICAL_ERROR`) to `IDatabase`?

File: src/Databases/MySQL/MaterializedMySQLSyncThread.cpp [hunk 3]
```diff
@@ -759,15 +760,9 @@ void MaterializedMySQLSyncThread::executeDDLAtomic(const QueryEvent & query_even
...
-bool MaterializedMySQLSyncThread::isMySQLSyncThread()
-{
-    return getThreadName() == MYSQL_BACKGROUND_THREAD_NAME;
-}
-
 void MaterializedMySQLSyncThread::setSynchronizationThreadException(const std::exception_ptr & exception)
 {
-    auto db = DatabaseCatalog::instance().getDatabase(database_name);
-    DB::setSynchronizationThreadException(db, exception);
+    std::static_pointer_cast<DatabaseMaterializedMySQL>(DatabaseCatalog::instance().getDatabase(database_name))->setException(exception);
```
Comment (tavplubix) [inline src/Databases/MySQL/MaterializedMySQLSyncThread.cpp:765]: Maybe use assert_cast?

File: src/Databases/DatabaseFactory.cpp (hunk 1 above)
Comment (stigsb) [reply src/Databases/DatabaseFactory.cpp:214]: Done.

File: src/Databases/MySQL/MaterializedMySQLSyncThread.cpp (hunk 3 above)
Comment (stigsb) [reply src/Databases/MySQL/MaterializedMySQLSyncThread.cpp]: Fixed.

File: src/Databases/IDatabaseReplicating.h (hunk 2 above)
Comment (stigsb) [reply src/Databases/IDatabaseReplicating.h]: Fixed.

**Review by tavplubix — APPROVED** [review]

**Comment by tavplubix** [comment]
Unrelated failures:
Stress test (thread) - #31423
Integration tests were broken in master

Related failures:
ClickHouse special build check
Stress tests (debug, actions) - can be reproduced with `ClickHouse/tests$ ./clickhouse-test  --db-engine="Replicated('/test/repl/db/123', 's1', 'r1')" 01185_create_or_replace_table`

**Comment by tavplubix** [comment]
`ClickHouse special build check` and `Stress test (debug, actions)` failures are related to changes in the PR, @stigsb, could you please fix it?

**Comment by stigsb** [comment]
Yes, I'll look at it today.

**Comment by stigsb** [comment]
@tavplubix looking at the remaining test failures:

"Fast test" seems to fail because the Raft port is already taken (could be a recent process exiting uncleanly?)

"Stress test (thread, actions)" has a failing test script, but I can't make out why from the logs. Any ideas @tavplubix ?

**Comment by tavplubix** [comment]
Tests failures are unrelated to changes

**Pull request merged**